src/main/java/com/rackspacecloud/metrics/rollup/serdes/PojoSerde.java

```diff
@@ -0,0 +1,122 @@
+package com.rackspacecloud.metrics.rollup.serdes;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This class creates Serializer and Deserializer for the given POJO type.
+ * @param <T> POJO type for the serializer and deserializer.
+ */
+public class PojoSerde<T> implements Serde<T> {
+    Class<T> pojoType;
+    Serializer<T> pojoSerializer;
+    Deserializer<T> pojoDeserializer;
+
+    public PojoSerde(){
+        throw new NotImplementedException();
+    }
+
+    public PojoSerde(Class<T> pojoType) {
+        this.pojoType = pojoType;
+        this.pojoSerializer = new PojoSerializer<>();
+        this.pojoDeserializer = new PojoDeserializer<>(pojoType);
+    }
+
+    @Override
+    public void configure(Map<String, ?> map, boolean isKey) {
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public Serializer<T> serializer() {
+        return this.pojoSerializer;
+    }
+
+    @Override
+    public Deserializer<T> deserializer() {
+        return this.pojoDeserializer;
+    }
+
+    static class PojoSerializer<T> implements Serializer<T> {
```

Comment:
Was there a reason you didn't use the ones provided by Spring Kafka

https://docs.spring.io/spring-kafka/docs/2.2.3.RELEASE/reference/html/_reference.html#serdes